nodeup/pkg/model/kube_apiserver_test.go

```diff
@@ -70,53 +71,63 @@ type fakeKeyStore struct {
 	T *testing.T
 }
 
+var _ fi.Keystore = &fakeKeyStore{}
+
 func (k fakeKeyStore) FindKeypair(name string) (*pki.Certificate, *pki.PrivateKey, bool, error) {
-	panic("implement me")
+	panic("fakeKeyStore does not implement FindKeypair")
 }
 
 func (k fakeKeyStore) CreateKeypair(signer string, name string, template *x509.Certificate, privateKey *pki.PrivateKey) (*pki.Certificate, error) {
-	panic("implement me")
+	panic("fakeKeyStore does not implement CreateKeypair")
 }
 
 func (k fakeKeyStore) StoreKeypair(id string, cert *pki.Certificate, privateKey *pki.PrivateKey) error {
-	panic("implement me")
+	panic("fakeKeyStore does not implement StoreKeypair")
 }
 
 func (k fakeKeyStore) MirrorTo(basedir vfs.Path) error {
-	panic("implement me")
+	panic("fakeKeyStore does not implement MirrorTo")
 }
 
-func (k fakeKeyStore) FindCertificatePool(name string) (*fi.CertificatePool, error) {
-	panic("implement me")
+type fakeCAStore struct {
+	fakeKeyStore
```

Comment:
Since you're changing all existing references to `fakeKeyStore` to `fakeCAStore`, why keep `fakeKeyStore` separate?

Reply:
I'm not entirely happy with the dual CAStore / KeyStore structure, and I'm hoping we can clean it up over time.  I didn't want to add another thing that would need to be untangled later.

Reply:
The only thing that takes just a `KeyStore` are `BuildKubecfg` and `fi.Context`. One thing using the field in `fi.Context` upcasts it to a `VFSCAStore`. (The other is `fitasks.Keypair`.) I don't think the interface split is well-designed.

Reply:
I do agree with that - it has resisted my attempts at making them cleaner (thus far)

Reply:
It comes down to sharing `fi.Context` between nodeup and cloudup. Two of the three methods in `KeyStore` are only needed by specific cloudup tasks.